sdk/python/kfp/dsl/pipeline_task.py

```diff
@@ -190,13 +191,20 @@ def _execute_locally(self, args: Dict[str, Any]) -> None:
         from kfp.local import task_dispatcher
 
         if self.pipeline_spec is not None:
-            raise NotImplementedError(
-                'Local pipeline execution is not currently supported.')
-
-        self._outputs = task_dispatcher.run_single_task(
-            pipeline_spec=self.component_spec.to_pipeline_spec(),
-            arguments=args,
-        )
+            self._outputs = pipeline_orchestrator.run_local_pipeline(
+                pipeline_spec=self.pipeline_spec,
+                arguments=args,
+            )
+        elif self.component_spec is not None:
```

Comment:
I think this LOC should not be hit for the case of a pipeline in a pipeline, since [`execute_locally`](https://github.com/kubeflow/pipelines/blob/8c7b5b2bf56beef42511bf640d35b2c040389cc9/sdk/python/kfp/dsl/pipeline_task.py#L182) would be `False` (passed in [here](https://github.com/kubeflow/pipelines/blob/8c7b5b2bf56beef42511bf640d35b2c040389cc9/sdk/python/kfp/dsl/base_component.py#L104)). For condition and iteration `DagSpec`s, there is no associated `PipelineTask` so this LOC would also not be hit.

Is there another case you had in mind?

In my WIP implementation of pipeline-in-pipeline local execution, I've also not hit an issue here.